File: head_to_offer/48/hto_48.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) == 0:
            return 0
        if len(s) == 1:
            return 1
        res = 0
        i = 0
        j = 1
        while j < len(s):
            word = s[i:j]
            if s[j] in word:
                res = max(res, j - i)
                i = s.find(s[j], i, j) + 1
                j = i + 1
            else:
                res = max(res, j - i + 1)
                j += 1
        return res

File: head_to_offer/48/hto_48.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        size_t len = s.size();
        if (s.empty()) return 0;
        if (len == 1) return 1;
        int res = 0;
        int i = 0, j = 1;
        string word;
        while (j < len) {
            word = s.substr(i, j - i);
			if (word.find(s[j]) != string::npos) {
				res = max(res, j - i);
				i = s.find(s[j], i) + 1;
				j = i + 1;
            } else {
                res = max(res, j - i + 1);
                j++;
            }
        }
        return res;
    }
};